### PR TITLE
Add `ActivityExtractionParamSet` table and fix Nikon scan duration calculation

### PR DESCRIPTION
**Pull request by tdincer**

### CODE REVIEW

File: element_calcium_imaging/imaging.py [hunk 1]
```diff
@@ -1024,48 +1015,116 @@ class ActivityExtractionMethod(dj.Lookup):
...
+            pname = q_param.fetch1("activity_extraction_paramset_idx")
+            if (
+                pname == activity_extraction_paramset_idx
+            ):  # If the existed set has the same name: job done
+                return
+            else:  # If not same name: human error, trying to add the same paramset with different name
+                raise dj.DataJointError(
+                    "The specified param-set already exists - name: {}".format(pname)
+                )
+        else:
```
Comment (dimitri-yatsenko) [inline element_calcium_imaging/imaging.py:1060]: This functionality seems redundant. I recommend removing it.

**Comment by kabilar** [comment]
Should we close #67?

**Comment by kabilar** [comment]
After second thought, please:
1. Submit the Nikon fix in a separate pull request which I will merge first.
2. We can then make a release of version 0.1.0.
3. Then we can merge this backwards incompatible change to the `Activity` table.
4. We can change the version to 0.2.0 and make a new release.

**Comment by tdincer** [comment]
> #67

Linking it to this PR.

File: element_calcium_imaging/imaging.py [hunk 2]
```diff
@@ -1024,48 +1015,116 @@ class ActivityExtractionMethod(dj.Lookup):
...
+                * ActivityExtractionParamSet
+                * ProcessingParamSet.proj("processing_method")
+                & 'processing_method in ("suite2p")'
+                & 'extraction_method = "FISSA"'
+            ).proj()
+        )
+        return (
+            suite2p_key_source.proj()
+            + caiman_key_source.proj()
+            + fissa_key_source.proj()
```
Comment (kabilar) [inline element_calcium_imaging/imaging.py:1119]: If I am understanding this correctly, this key source would in effect return `Fluorescence * ActivityExtractionParamSet`.  Alternatively, should we add an `ActivityTask` table?

**Review by kabilar — CHANGES_REQUESTED** [review]
Thanks @tdincer!  Great work.  I have listed a few suggestions.

File: element_calcium_imaging/imaging.py (hunk 2 above)
Comment (kabilar) [reply element_calcium_imaging/imaging.py:1119]: Hi @ttngu207, what are your thoughts on this suggestion?

File: element_calcium_imaging/imaging_no_curation.py [hunk 3]
```diff
@@ -1026,18 +1081,17 @@ def make(self, key):
...
+                                "activity_type": "spike",
                                 "activity_trace": spks,
                             }
                         )
 
-                self.insert1(key)
-                self.Trace.insert(spikes)
-        elif method == "caiman":
+                self.insert1(key)  # FIX THIS
+                self.Trace.insert(spikes)  # FIX THIS
```
[the remark this answers is not in the document]
Comment (tdincer) [reply element_calcium_imaging/imaging_no_curation.py:1090]: Something was wrong but not anymore. Those comments need to be removed.

File: element_calcium_imaging/imaging.py (hunk 2 above)
Comment (dimitri-yatsenko) [reply element_calcium_imaging/imaging.py]: union `+` is a very inefficient query and should not be used.

**Pull request closed without merging**